docs/proposals/PROPOSAL_E.md

```diff
@@ -184,6 +189,36 @@ For registries that do not support the `referrers` API, a tag MUST be pushed for
 - Periodic garbage collection may be performed by clients pushing new referrers, deleting stale referrers that have been replaced with newer versions, and tags that no longer point to an accessible manifest.
 - Clients can verify the registry does not support the `referrers` API by querying the API and checking for a 404.
 
+#### Client Expectations
+
+##### Creating Artifacts
+
+- For portability, clients should generate artifacts using image-spec until all registries where the artifact is pushed to have been upgraded.
+- The descriptor of the manifest this artifact refers to should be set in the `refers` field of the manifest.
+- Other metadata should be included in the manifest annotations.
+- When pushing, the registry should be checked for the referrers API support.
+- If the referrers API is not available, the artifact should be pushed with a tag using the digest schema. Tags with the digest schema should not be pushed if the referrers API is available.
+
+##### Pulling Artifacts
+
+To pull artifacts that reference an existing manifest:
+
+- Clients requests artifacts associated with a manifest first query the referrers API using the digest of the requested manifest.
+- If the request fails, clients must fall back to listing tags in the repository and pull manifests for each tag with the matching `<alg>` and `<ref>` prefix.
+- Clients then pull any artifact matching their criteria from the API response or the list generated from the tag query.
+
+##### Copying Images
+
+- Tooling that copies images between registries should recursively query for referrers.
+- Copying an artifact that refers to an image uses the same pull and push steps described above and should support both existing registries and registries with the `referrers` API on both the pull and push.
+
+##### Deleting Manifests
+
+For managing existing registries without the `referrers` API:
+
+- Client tooling that deletes manifests should also delete digest tags that reference that manifest.
+- Client tooling should periodically check for dangling digest tags that refer to missing manifest, and prune those tags.
```

Comment:
It might be worth considering the scenario where folks want to keep some referrers without dropping them. This prune recommendation might mean they should tag differently to keep things around.

Reply:
I'm leaning towards keeping the scope in the proposal to what's needed to manage an artifact that refers to another manifest. The fear is if we start documenting some other scenarios, we'll be asked to document them all. But to your point, yes, an artifact can be tagged and pulled directly.

Reply:
Can I ask for the "should" to be changed to "could" or "may"? As we start to think about how this will land as official spec changes, words like "SHOULD" get new meaning, and I'd like to avoid recommending with the force that a "SHOULD" brings with it.

Reply:
@imjasonh Done.